xlsform_app/views.py

```diff
@@ -15,18 +21,29 @@
 from pyxform import xls2json
 from pyxform.utils import has_external_choices
 from pyxform.xls2json_backends import sheet_to_csv
+from pyxform.validators import odk_validate
 
 DJANGO_TMP_HOME = os.environ['DJANGO_TMP_HOME']
+DJANGO_PERSISTENT_HOME = os.environ['DJANGO_PERSISTENT_HOME']
 
 class UploadFileForm(forms.Form):
     file = forms.FileField()
 
+class Persistence(Enum):
+    NONE = 'none'
+    ODK_ONLY = 'odk-only'
+    PUBLIC_ANONYMOUS = 'public-anonymous'
+    PUBLIC = 'public'
```

Comment:
Instead of `Persistence` for the enum, I'd go with `ViewableBy` or something like that. Then the enum would be `NOBODY`, `ODK_ONLY`. The variable name that stores that value would then be `viewable_by`.

I'd drop `PUBLIC` and `PUBLIC_ANONYMOUS` because we aren't using them and I don't like seeing dead code.

Persistence for the home directory and methods on the actual persisting of files is fine though.

Reply:
As discuss on slack, I have removed this logic. Now all uploaded files are saved in DJANGO_PERSISTENT_HOME.

Files uploaded via API are saved in `web-forms` subdirectory and files uploaded via HTML Form are saved in `enketo` subdirectory.

XML files generated via pyxform conversion process continue to be saved in DJANGO_TMP_HOME. 

In this PR, I have added ability to preview XML files as well, so uploaded XML files are first saved in PERSISTENT_HOME and then copied to TMP_HOME so that they can be accessed via `/downloads/...` endpoint